### PR TITLE
EIP-1153: Transient Storage tests

### PR DESCRIPTION
**Pull request by moodysalem**

### CODE REVIEW

**Comment by marioevz** [comment]
I haven't looked at all the tests but one thing that came to mind when hearing about this EIP is that the clients need to implement a new data structure that might be ephemeral but we might still need to test the maximum number of transient elements it can realistically contain within the span of a single transaction, just to make sure that this structure does not break.

**Comment by moodysalem** [comment]
> I haven't looked at all the tests but one thing that came to mind when hearing about this EIP is that the clients need to implement a new data structure that might be ephemeral but we might still need to test the maximum number of transient elements it can realistically contain within the span of a single transaction, just to make sure that this structure does not break.

Mentioned in the EIP security considerations: https://eips.ethereum.org/EIPS/eip-1153#security-considerations

And it should be tested in the file `src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml`

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      code: "0x"
+      nonce: 0
+      storage: {}
+
+  transaction:
+    data:
+      - data: :abi doCall(uint) 0x200
+        accessList: []
+    gasLimit:
+      - "400000"
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: should probably be much more than 400k, like 30m gas, and `0x200` -> `0x30d40` (200k)

**Comment by winsvega** [comment]
can this EIP be activated by command `Merge+1153`?

**Comment by snreynolds** [comment]
> can this EIP be activated by command `Merge+1153`?

updated, thank you. can now run them off the develop branch of retesteth with the following commands

GeneralStateTests:

```
/dretesteth.sh -t GeneralStateTests/stEIP1153 -- \
 --testpath <path to tests>  --filltests
```

BlockchainTests:

```
./dretesteth.sh -t BlockchainTests/ValidBlocks/bcEIP1153 -- \
 --testpath <path to tests>  --filltests
```

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml (hunk 1 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: fixed here https://github.com/ethereum/tests/pull/1091/commits/d1f8ba85bc0c2c4d150aa7390bc56a7913f0a6e5

**Comment by snreynolds** [comment]
@marioevz Would love to get any other thoughts you might have for these tests

**Comment by winsvega** [comment]
ready to merge? 
for the blockchain test,  t8n tool does not construct a blockchain. so instead it is treated like a chain of state tests. (single state in, single state out.) 
the bc test produced here is valid because you don't expect the cache to be present in the next block. 
however if you wanted to make a test that it does present such thing won't work with t8n.

the client that could run it is besu once it has 1153 support.

File: src/GeneralStateTestsFiller/stEIP1153/03_tloadAfterStoreIs0Filler.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  pre:
+    000000000000000000000000000000000000000A:
+      balance: 1000000000000000000
+      nonce: 0
+      code: |
+        :yul {
+          verbatim_2i_0o(hex"b4", 0, 30)
+          let val := verbatim_1i_1o(hex"b3", 1)
+          sstore(0, val)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/03_tloadAfterStoreIs0Filler.yml]: I think it would be a good idea in this test to sstore into a different location than tstore, just to test at the end that the tstore didn't somehow end up being written to real storage.

E.g.
```
tstore(0, 1)
sstore(1, <some test result>)
expect:
0x00: 0 # With this, we are testing that tstore didn't get stored in real storage
0x01: <some test result>
```

File: src/GeneralStateTestsFiller/stEIP1153/04_tloadAfterCallFiller.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          sstore(1, success)
+        }
+      storage: {}
+    000000000000000000000000000000000000000B:
+      balance: 1000000000000000000
+      nonce: 0
+      code: |
+        :yul {
+          let val := verbatim_1i_1o(hex"b3", 0)
+          sstore(0, val)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/04_tloadAfterCallFiller.yml]: Also `tstore` after this `sstore` with a different value than the calling contract, to test that the `tstore` doesn't modify calling contract transient storage.

File: src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            doReenter()
+          }
+
+          default {
+            sstore(2, 1)
+          }
+
+          function doStoreAndRevert() {
+            verbatim_2i_0o(hex"b4", 0, 11)
+            revert(0, 0)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml]: Is revert the only method used throughout the tests?

If so, I think it would be valuable to add at least one test case with other types of failures such as:
- out of gas
- invalid opcode
- stack underflow

File: src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+          function storeTwiceAndRevert() {
+            verbatim_2i_0o(hex"b4", 0, 5)
+            verbatim_2i_0o(hex"b4", 0, 6)
+            revert(0,0)
+          }
+
+          function doReenter() {
+            verbatim_2i_0o(hex"b4", 0, 4)
+            mstore(0, hex"578290e5")
+            let success := call(0xffff, 0x000000000000000000000000000000000000000A, 0, 0, 32, 0, 0)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml]: Var name should be `fail`.

File: src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+          case 0x578290e5 { // storeTwiceAndRevert()
+            storeTwiceAndRevert()
+          }
+
+          default {}
+
+          function storeTwiceAndRevert() {
+            verbatim_2i_0o(hex"b4", 0, 5)
+            verbatim_2i_0o(hex"b4", 0, 6)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml]: Maybe also do `tstore(1, x)` and check that on the caller too.

File: src/GeneralStateTestsFiller/stEIP1153/11_tstoreDelegateCallFiller.yml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        :yul {
+          verbatim_2i_0o(hex"b4", 0, 1)
+          let success := delegatecall(0xffff, 0x000000000000000000000000000000000000000B, 0, 32, 0, 0)
+          let v := verbatim_1i_1o(hex"b3", 0)
+          sstore(0, success)
+          sstore(1, v)
+        }
+
+      storage: {}
+    000000000000000000000000000000000000000B:
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/11_tstoreDelegateCallFiller.yml]: Maybe add another contract that does `tstore` but reverts.

File: src/GeneralStateTestsFiller/stEIP1153/13_tloadStaticCallFiller.yml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+          }
+
+          case 0x611e535a { // tstore3()
+            tstore3()
+          }
+
+          default {}
+
+          function tstore3() {
+            verbatim_2i_0o(hex"b4", 0, 3)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/13_tloadStaticCallFiller.yml]: We are missing a test case where we `staticcall` and then `tload`, which can be verified by returning the value and then saving it on the caller contract.

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+            let t := call(0xffff, 0x000000000000000000000000000000000000000A, 0, 0, 64, 0, 0)
+            sstore(0, t) // should revert
+
+            let v := verbatim_1i_1o(hex"b3", 0)
+            sstore(1, v)
+          }
+
+          function doNStores(n) {
+            for { let i := 0 } lt(i, n) { i := add(i, 1) } {
+              verbatim_2i_0o(hex"b4", i, 48)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: Could be interesting to add a similar test to this but skipping some slots in between the stored values.

Also one where the slots are written out of order:
$$0, 2^{256}-1, 1, 2^{256}-2, ...$$

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+          }
+
+          default {
+            sstore(0, 5)
+          }
+          
+          function doCall(n) {
+            mstore(0, hex"f7d6c0eb") // doNStores(n)
+            mstore(32, n)
+            let t := call(0xffff, 0x000000000000000000000000000000000000000A, 0, 0, 64, 0, 0)
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: I think the `0xffff` might be a very low value considering that the transaction in total has 30,000,000 gas.
15 seconds for 65535 gas in operations is in my opinion excessively long.

File: src/GeneralStateTestsFiller/stEIP1153/16_tloadGasFiller.yml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+      code: |
+        :yul {
+          let extraOpcodesGas := 5
+
+          // calculate warm sload gas
+          let s := sload(5)
+          let g1 := gas()
+          let s2 := sload(5)
+          let g2 := gas()
+          sstore(0, sub(sub(g1, g2), extraOpcodesGas))
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/16_tloadGasFiller.yml]: I don't think it's worth it to include the gas costs for `sload` here, since this is an entirely different test suite, if in the future for some reason the `sload` gas cost changes, this test will fail.

File: src/GeneralStateTestsFiller/stEIP1153/17_tstoreGasFiller.yml [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+      nonce: 0
+      code: |
+        :yul {
+          let extraOpcodes := 8
+
+          // calculate warm dirty sstore gas
+          sstore(5, 1)
+          let g1 := gas()
+          sstore(5, 2)
+          let g2 := gas()
```
Comment (marioevz) [inline src/GeneralStateTestsFiller/stEIP1153/17_tstoreGasFiller.yml]: Same observation as with previous test.

**Review by marioevz — CHANGES_REQUESTED** [review]
I think it looks good overall.

The DoS test seemed a bit worrying to me since it was not that much gas and can take way too long.

Also there should be more tests using more transient slots than 0, almost all tests only check the zero slot, so we should check other interesting keys too.

**Comment by holiman** [comment]
> won't work with t8n.

Really? I think it should, but if it doesn't, we need to get that fixed.

**Comment by holiman** [comment]
Seems to work
```
[user@work evm]$ ./evm t8n --state.fork=Merged+1153 --input.alloc=./testdata/23/alloc.json --input.txs=./testdata/24/txs.json --input.env=./testdata/25/env.json
INFO [11-30|22:04:22.208] rejected tx                              index=0 hash=92ea4a..2ee941 from=0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B error="nonce too high: address 0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B, tx: 172 state: 0"
INFO [11-30|22:04:22.209] rejected tx                              index=1 hash=16b1d9..c4f1da from=0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B error="nonce too high: address 0xa94f5374Fce5edBC8E2a8697C15331677e6EbF0B, tx: 173 state: 0"
INFO [11-30|22:04:22.209] Trie dumping started                     root=b7360b..e3458f
INFO [11-30|22:04:22.209] Trie dumping complete                    accounts=2 elapsed="51.703µs"
INFO [11-30|22:04:22.209] Wrote file                               file=alloc.json
INFO [11-30|22:04:22.209] Wrote file                               file=result.json
```
As opposed to e.g.: 
```
[user@work evm]$ ./evm t8n --state.fork=Merged+1199 --input.alloc=./testdata/23/alloc.json --input.txs=./testdata/24/txs.json --input.env=./testdata/25/env.json
ERROR(3): failed constructing chain configuration: syntax error, invalid eip number 1199

```

File: src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml (hunk 4 above)
Comment (winsvega) [reply src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml]: Yes. One thing we were working on are templates for new opcodes - a basic series of scenarious to cover for any new opcode. 

Like oogs, reverts, underflow, static context... 

@qbzzt I think you even did a template generator for some of that

**Comment by holgerd77** [comment]
> The DoS test seemed a bit worrying to me since it was not that much gas and can take way too long.

Has this comment from @marioevz been picked up since it might be important to dig in further here and have a somewhat more solid picture? Mario, could you expand a bit on this by citing the numbers (gas used, runtime, machine) to get some first impression here?

@holiman are you able to give a first-round judgement on this how critical this is/might be respectively how much worth to be deeper investigated?

**Comment by holiman** [comment]
Re the Dos-issue. The description says: `CALL(self, ...), TSTORE(x, y), TSTORE(x + 1, y), TSTORE(x + 2, y), ... ,TSTORE(x + n, y), REVERT, TLOAD(x)`. So essentially, it just does a lot of `TSTORE` ? 

Sure, it might be something we need to look into (it ought not be a problem, but implementations can always be surprising) -- but I don't see why it would be a blocking issue for _this_ pr.

File: src/GeneralStateTestsFiller/stEIP1153/03_tloadAfterStoreIs0Filler.yml (hunk 2 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/03_tloadAfterStoreIs0Filler.yml]: fixed here https://github.com/ethereum/tests/pull/1091/commits/9a54b73adf145c945bfb72d63513e707b4526131 we now expect storage slot 0 to be 0xffff, the same at the start of the txn. And we now expect storage slot 1 to be 0 which is the loaded value from an unused key

File: src/GeneralStateTestsFiller/stEIP1153/04_tloadAfterCallFiller.yml (hunk 3 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/04_tloadAfterCallFiller.yml]: fixed https://github.com/ethereum/tests/pull/1091/commits/2aace0267dce263e8439d44e10cef18397c24e8a

File: src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml (hunk 4 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/08_revertUndoesTransientStoreFiller.yml]: Added some oog tests here
https://github.com/ethereum/tests/pull/1091/commits/2bcc343480ed47b2dd6cd0bf1957a95aae3447bc
https://github.com/ethereum/tests/pull/1091/commits/c86bd00f9ff7f6938b48f7a3767c18d48e54b6ae

File: src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml (hunk 6 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml]: added storing/loading from transient slot 1 https://github.com/ethereum/tests/pull/1091/commits/285c951c37e8b9cbcca24249b7bbc8d495a7b97f

File: src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml (hunk 5 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/09_revertUndoesAllFiller.yml]: good catch thanks https://github.com/ethereum/tests/pull/1091/commits/285c951c37e8b9cbcca24249b7bbc8d495a7b97f

File: src/GeneralStateTestsFiller/stEIP1153/11_tstoreDelegateCallFiller.yml (hunk 7 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/11_tstoreDelegateCallFiller.yml]: adds revert after delegatecall https://github.com/ethereum/tests/pull/1091/commits/31319495e03410614c38351d6498aa7d6c393435

File: src/GeneralStateTestsFiller/stEIP1153/13_tloadStaticCallFiller.yml (hunk 8 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/13_tloadStaticCallFiller.yml]: added tload https://github.com/ethereum/tests/pull/1091/commits/11ffdeb108d5379fa9f4bd45d61aba17d677bea1

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml (hunk 10 above)
Comment (moodysalem) [reply src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: should probably change to `call(gas(), ...`

Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: updated https://github.com/ethereum/tests/pull/1091/commits/c07c1485a9b43874b9ee106ebeb18fdaef98836e

File: src/GeneralStateTestsFiller/stEIP1153/16_tloadGasFiller.yml (hunk 11 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/16_tloadGasFiller.yml]: makes sense, fixed https://github.com/ethereum/tests/pull/1091/commits/947ebcaeb3e33266f7af456722057536595db53a & https://github.com/ethereum/tests/pull/1091/commits/2556022df9340b4fe1911a64478885b26f22fb9b

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml (hunk 9 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: added some out of order stores here https://github.com/ethereum/tests/commit/75e479fb379d84db383fa51247e654211aad5fe3

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+  pre:
+    000000000000000000000000000000000000000A:
+      balance: 1000000000000000000
+      nonce: 0
+      code: |
+        :yul {
+          switch selector()
+
+          case 0x883264e8 { // doCall(uint)
+            let n:= div(calldataload(32), 0x100000000000000000000000000000000000000000000000000000000)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: i don't think you want a div here

File: src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+          }
+
+          default {
+            sstore(0, 5)
+          }
+          
+          function doCall(n) {
+            mstore(0, hex"f7d6c0eb") // doNStores(n)
+            mstore(32, n)
+            let t := call(gas(), 0x000000000000000000000000000000000000000A, 0, 0, 64, 0, 0)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/15_tstoreCannotBeDosdFiller.yml]: could be more clear that it's self-calling if it's `address()` instead of `0x000000000000000000000000000000000000000A`

File: src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+  pre:
+    000000000000000000000000000000000000000A:
+      balance: 1000000000000000000
+      nonce: 0
+      code: |
+        :yul {
+          switch selector()
+
+          case 0x883264e8 { // doCall(uint)
+            let n:= div(calldataload(32), 0x100000000000000000000000000000000000000000000000000000000)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml]: unnecessary div here

File: src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+            mstore(32, n)
+            let t := call(gas(), 0x000000000000000000000000000000000000000A, 0, 0, 64, 0, 0)
+            sstore(0, t) // should revert
+
+            let v := verbatim_1i_1o(hex"b3", 0)
+            sstore(1, v)
+          }
+
+          function doNStores(n) {
+            let val := exp(2, 256)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml]: 2**256 should return 0 i expect

File: src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+              verbatim_2i_0o(hex"b4", i, 600)
+              let even := mod(i, 2)
+              let index
+              switch even 
+              case 0 {
+                index := i
+                verbatim_2i_0o(hex"b4", i, 600)
+              }
+              case 1 {
+                index := sub(val, i)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml]: this probably still works because of underflow

File: src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml [hunk 18]
```diff
@@ -0,0 +1,105 @@
...
+                 verbatim_2i_0o(hex"b4", index, 600)
+               }
+              default {}
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml]: no default needed

**Comment by winsvega** [comment]
please move tests to EIPTests folder in BlockchainTests and GeneralStateTests accordingly.

File: src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml (hunk 18 above)
Comment (snreynolds) [reply src/GeneralStateTestsFiller/stEIP1153/21_tstoreCannotBeDosdOOOFiller.yml]: addressed all the above https://github.com/ethereum/tests/pull/1091/commits/bc0dcbc292b8256d9566b05be2eccab60a2e73c9 &
https://github.com/ethereum/tests/pull/1091/commits/eb0a02818c20be023e6407928054d3478b51b899 & 
https://github.com/ethereum/tests/pull/1091/commits/fcd3575118cc621b2ada3f4d0b88070ea9f81263

File: src/GeneralStateTestsFiller/EIPTests/stEIP1153/15_tstoreCannotBeDosdFiller.yml [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+
+  pre:
+    000000000000000000000000000000000000000A:
+      balance: 1000000000000000000
+      nonce: 0
+      code: |
+        :yul {
+          switch selector()
+
+          case 0x883264e8 { // doCall(uint)
```
Comment (moodysalem) [inline src/GeneralStateTestsFiller/EIPTests/stEIP1153/15_tstoreCannotBeDosdFiller.yml:28]: bit strange how the selector is computed from `doCall(uint)` rather than `doCall(uint256)` which is how solidity would do it, and that `:abi doCall(uint)` on line 72 below takes the signature as a literal instead of parsing it

Comment (moodysalem) [reply src/GeneralStateTestsFiller/EIPTests/stEIP1153/15_tstoreCannotBeDosdFiller.yml:28]: in other words, this is not consistent with solidity. doesn't really matter, but thought i'd point it out. you also have a function selector computed from `keccak256('doNStores(n)')` which is what made me think of it (using `n` instead of the type in the signature)

**Review by winsvega — CHANGES_REQUESTED** [review]
git reset --soft HEAD~60

Can repack this into smaller number of commits. Like test srcs, test filled separate
We don't need history of editing and moving from folder to folder. 
And its ready to merge into EIPTests folder, then we can pr to make it better ib another thread. And eventually update to cancun or when it's gonna be accepted

**Comment by snreynolds** [comment]
> git reset --soft HEAD~60
> 
> Can repack this into smaller number of commits. Like test srcs, test filled separate We don't need history of editing and moving from folder to folder. And its ready to merge into EIPTests folder, then we can pr to make it better ib another thread. And eventually update to cancun or when it's gonna be accepted

@winsvega Can we close this in favor of https://github.com/ethereum/tests/pull/1194/files which squashes these commits and is updated against latest develop branch

**Comment by winsvega** [comment]
Finally ready to merge. Thanks for the contribution!
https://github.com/ethereum/tests/pull/1196
